numpy_challenge.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+
+if __name__ == "__main__":  # creatint 3 different numpy arrays
+    arr_1 = np.eye(3, dtype=int)
+    arr_2 = np.arange(9).reshape((3, 3))
+    arr_3 = np.fromstring('7 3', dtype=int, sep=' ')  # fav number of Sheldon Cooper
+
+
+def matrix_multiplication(a, b):  # a and b are matrices
+    return np.matmul(a, b)
+
+
+def multiplication_check(a):  # a is an array of matrices
+    for i in range(1, len(a)):
+        if np.shape(a[i])[0] != np.shape(a[i - 1])[1] or np.shape(a[i])[1] != np.shape(a[i - 1])[0]:
+            return False
+    return True
+
+
+def multiply_matrices(a):  # a is an array of matrices
+    matt = a[0]
+    for i in range(1, len(a)):
+        if np.shape(a[i])[0] != np.shape(a[i - 1])[1] or np.shape(a[i])[1] != np.shape(a[i - 1])[0]:
+            return None
+        matt = np.matmul(a[i], matt)
+    return matt
```

Comment:
Странный пример, но допустим, что у нас есть список из 1е4 матриц и мы их перемножаем. При этом последние две перемножится не могут. Получится, что мы зря потратили кучу времени на расчёт для предыдущих 1е4 - 1. Это следует проверить в самом начале.

К тому же у нас уже есть функция для проверки на переможаемость. Изобретать велосипед заново не надо

Нарушен порядок операций. Мы должны предыдущий результат умножать на следующую матрицу, а не наоборот

```suggestion
def multiply_matrices(a):  # a is an array of matrices
    if not multiplication_check(a):
        return None
    matt = a[0]
    for i in range(1, len(a)):
        matt = matrix_multiplication(matt, a[i])
    return matt
```